src/client/pages/matchPage/MatchPage.jsx

```diff
@@ -1,53 +1,118 @@
 import React from 'react';
-//import { useState } from 'react';
+import { useState, useEffect } from 'react';
+import { GET_BOOKS_URL } from '../../config';
+import { generateRandomInt } from '../../utils';
 
 import Footer from '../../components/Footer'
 
 import '../../styles/match.css'
 
 const MatchPage = () => {
-    // const [book, setBook] = useState({});
+    const [library, setLibrary] = useState({});
+    const [book, setBook] = useState({});
+    const [displayMatchCard, setDisplayMatchCard] = useState(true);
+    const [recommendations, setRecommendations] = useState([]);
+
+    const pickRandomBook = (library) => {
+        const generatedIndex = generateRandomInt(library.length - 1);
+        
+        return library[generatedIndex];
+    }
+    console.log(library);
+    const pickGroupFromDifferentGroup = (unwantedGroupId) => {
+        let bookToMatch;
+
+        do{
+            bookToMatch = pickRandomBook(library);
+        } while (bookToMatch.groupId === unwantedGroupId);
```

Comment:
The rare and fabled `do while`! Nice

Reply:
Thanks. It took me a sec to remember the syntax :')